prism_compile.c

```diff
@@ -3331,22 +3340,22 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
             }
         }
 
-        if (multi_write_node->rest) {
-            RUBY_ASSERT(PM_NODE_TYPE_P(multi_write_node->rest, PM_SPLAT_NODE));
-
-            pm_splat_node_t *rest_splat = ((pm_splat_node_t *)multi_write_node->rest);
-            if (rest_splat->expression) {
-                ADD_INSN2(ret, &dummy_line_node, expandarray, INT2FIX(0), INT2FIX(1));
-                PM_COMPILE(rest_splat->expression);
+        if (rights->size) {
```

Comment:
This logic could be split in a few ways. Let me know if there's a way that seems cleaner to you. There are basically three cases with slightly diff behavior: (rights && rest), rights, rest

Reply:
This seems fine!